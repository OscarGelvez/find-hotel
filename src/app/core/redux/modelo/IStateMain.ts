export interface IStateMain {
  isLoading: boolean;
}
