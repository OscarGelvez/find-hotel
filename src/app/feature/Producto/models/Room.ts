export interface IRoom {
  /**
   * Identificador del hotel
   */
  id: number;
  /**
   * Objeto hotel al cual pertenece la habitación
   */
  hotel: IHotel;
  /**
   * Valor de la habitación
   */
  value: number;
  /**
   * Indica si tiene o no aire acondicionado
   */
  ac: string;
  /**
   * Indica si cuenta o no con parquedadero
   */
  parking: string;
  /**
   * Fecha inicial de disponibilidad
   */
  available_from: string;
  /**
   * Fecha final de disponibilidad
   */
  available_until: string;
  /**
   * Indica si cuenta con conexión wifi
   */
  wifi: string;
  /**
   * Capacidad de personas por habitación
   */
  capacity: number;
  /**
   * Descripción general de la habitación
   */
  description: string;
}

export interface IHotel {
  /**
   * Identificador del hotel
   */
  id: number;
  /**
   * Nombre del hotel
   */
  name: string;
  /**
   * Dirección y/o ubicacion del hotel
   */
  address: string;
  /**
   * Cantidad de estrellas del hotel
   */
  stars: number;
  /**
   * Descripción general
   */
  description: string;
  /**
   * Puntaje de calificación dado por usuarios
   */
  score: number;
}
