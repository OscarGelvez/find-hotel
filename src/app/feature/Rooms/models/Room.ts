export interface IRoom {
  /**
   * Identificador del hotel
   */
  id: number;
  /**
   * Título de la habitación
   */
  title: string;
  /**
   * Imagen descriptiva de la habitación
   */
  image: string;
  /**
   * Objeto hotel al cual pertenece la habitación
   */
  hotel: IHotel;
  /**
   * Valor de la habitación
   */
  value: string;
  /**
   * Indica si tiene o no aire acondicionado
   */
  ac: string;
  /**
   * Indica si cuenta o no con parquedadero
   */
  parking: string;
  /**
   * Fecha inicial de disponibilidad
   */
  available_from: string;
  /**
   * Fecha final de disponibilidad
   */
  available_until: string;
  /**
   * Indica si cuenta con conexión wifi
   */
  wifi: string;
  /**
   * Capacidad de personas por habitación
   */
  capacity: number;
  /**
   * Descripción general de la habitación
   */
  description: string;
}

export interface IHotel {
  /**
   * Identificador del hotel
   */
  id: number;
  /**
   * Nombre del hotel
   */
  name: string;
  /**
   * Dirección y/o ubicacion del hotel
   */
  address: string;
  /**
   * Cantidad de estrellas del hotel
   */
  stars: number;
  /**
   * Descripción general
   */
  description: string;
  /**
   * Puntaje de calificación dado por usuarios
   */
  score: number;
}

export interface IFieldsForm {
  /**
   * Fecha inicio de reservación
   */
  fromDate?: string;
  /**
   * Fecha final de reservación
   */
  untilDate?: string;
  /**
   * Cantidad de personas en habitación
   */
  capacity?: string;
  /**
   * Habitación con aire acondicionado
   */
  withAir?: boolean;
  /**
   * Habitación con parqueadero propio
   */
  withParking?: boolean;
}

export interface IFieldsFormBooking {
  /**
   * Nombre de ususario que hace la reserva
   */
  name: string;
  /**
   * Correo electrónico de quien hacer la reserva
   */
  email: string;
  /**
   * Tipo de identificación
   */
  identification_type: string;
  /**
   * Número de identificación
   */
  identification: string;
}

export interface IBookingData {
  /**
   * Objeto con información de la habitación seleccionada
   */
  roomData: IRoom;
  /**
   * Objeto con la información el usuari oque reserva
   */
  bookData: IFieldsFormBooking;
}

export interface IFieldsFormFindBooking {
  /**
   * Correo electrónico de quien hacer la reserva
   */
  email: string;
  /**
   * Tipo de identificación
   */
  identification_type: string;
  /**
   * Número de identificación
   */
  identification: string;
}
