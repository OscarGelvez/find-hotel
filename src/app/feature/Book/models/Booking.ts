import { IRoom } from 'app/feature/Rooms/models/Room';

export interface IFieldsFormBooking {
  /**
   * Nombre de ususario que hace la reserva
   */
  name: string;
  /**
   * Correo electrónico de quien hacer la reserva
   */
  email: string;
  /**
   * Tipo de identificación
   */
  identification_type: string;
  /**
   * Número de identificación
   */
  identification: string;
  /**
   * Fecha de ingreso al hotel
   */
  fromDateBook: string;
  /**
   * Fecha de salida del hotel
   */
  untilDateBook: string;
}

export interface IBooking {
  /**
   * Objeto con información de la habitación seleccionada
   */
  roomData: IRoom;
  /**
   * Objeto con la información el usuari oque reserva
   */
  bookData: IFieldsFormBooking;
  /**
   * Identificador en base de datos
   */
  id?: number;
}

export interface IFieldsFormFindBooking {
  /**
   * Correo electrónico de quien hacer la reserva
   */
  email: string;
  /**
   * Tipo de identificación
   */
  identification_type: string;
  /**
   * Número de identificación
   */
  identification: string;
}
