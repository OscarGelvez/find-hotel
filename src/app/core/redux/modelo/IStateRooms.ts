import { IRoom } from 'app/feature/Rooms/models/Room';

export interface IStateRooms {
  listRooms: IRoom[];
  selectedRoomId: number;
}
