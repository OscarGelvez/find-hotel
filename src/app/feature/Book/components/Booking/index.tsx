import { fieldsFormFindBooking } from 'app/feature/Rooms/models/Room';
import { Button } from 'app/shared/components/Button';
import { Field, Form, Formik, FormikHelpers } from 'formik';
import * as PropTypes from 'prop-types';
import * as React from 'react';
import * as Yup from 'yup';

interface BookingDetail {
  onFindBooking: (findBookingData: fieldsFormFindBooking) => void;
}

const initialValues = {
  email: '',
  identification_type: '',
  identification: '',
};

const validationSchema = Yup.object().shape<fieldsFormFindBooking>({
  email: Yup.string().email('Email no válido').required('Necesitamos tu email'),
  identification_type: Yup.string().required(
    'Necesitamos tu tipo de identificación'
  ),
  identification: Yup.string().required('Necesitamos tu identificación'),
});

export const BookingDetail: React.FC<BookingDetail> = ({ onFindBooking }) => {
  const handleSubmit = (
    values: fieldsFormFindBooking,
    { resetForm }: FormikHelpers<fieldsFormFindBooking>
  ) => {
    onFindBooking({
      email: values.email,
      identification_type: values.identification_type,
      identification: values.identification,
    });
    resetForm();
  };
  return (
    <>
      <div data-testid="find-rooms" className="card-rooms">
        <div className="row">
          <div className="card filter-area ">
            <div className="col">
              <Formik
                initialValues={initialValues}
                validationSchema={validationSchema}
                onSubmit={handleSubmit}
              >
                {({ errors, touched }) => (
                  <Form data-testid="form-find-rooms">
                    <div className="form-group row mb-3 mt-3">
                      <div className=" col-12 col-md-4">
                        <label
                          htmlFor="email"
                          className="col-12 col-form-label"
                        >
                          Correo electrónico
                        </label>
                        <Field
                          name="email"
                          type="email"
                          className="form-control col-12"
                          placeholder="eje: joe@email.com"
                        />
                        {errors.email && touched.email ? (
                          <small className="text-danger">{errors.email}</small>
                        ) : null}
                      </div>
                      <div className="col-12 col-md-4">
                        <label className="pr-2" htmlFor="identification_type">
                          Tipo de identificación
                        </label>
                        <Field
                          as="select"
                          className=" d-block w-100"
                          aria-label=".form-select-sm example"
                          id="identification_type"
                          name="identification_type"
                        >
                          <option value="">Seleccione</option>
                          <option value="1">Cédula de ciudadanía</option>
                          <option value="2">Cédula de extranjeria</option>
                          <option value="3">Nit</option>
                          <option value="4">Pasaporte</option>
                        </Field>
                        {errors.identification_type &&
                        touched.identification_type ? (
                          <small className="text-danger">
                            {errors.identification_type}
                          </small>
                        ) : null}
                      </div>
                      <div className=" col-12 col-md-4">
                        <label className="pr-2" htmlFor="identification">
                          Identificación
                        </label>
                        <Field
                          name="identification"
                          className="form-control"
                          placeholder="eje: 60830147"
                        />
                        {errors.identification && touched.identification ? (
                          <small className="text-danger">
                            {errors.identification}
                          </small>
                        ) : null}
                      </div>
                    </div>
                    <div className="row text-center">
                      <Button type="submit" className="btn ">
                        Buscar
                      </Button>
                    </div>
                  </Form>
                )}
              </Formik>
            </div>
          </div>
        </div>
      </div>
    </>
  );
};

BookingDetail.propTypes = {
  // msg: PropTypes.string.isRequired,
};
