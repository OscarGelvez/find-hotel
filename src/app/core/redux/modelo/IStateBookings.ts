import { IBooking } from '../../../feature/Book/models/Booking';

export interface IStateBookings {
  listBookings: IBooking[];
  deleteBookingId: number;
}
