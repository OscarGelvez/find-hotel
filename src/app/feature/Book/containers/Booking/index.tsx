import {
  IBooking,
  IFieldsFormFindBooking,
} from 'app/feature/Book/models/Booking';
import * as PropTypes from 'prop-types';
import * as React from 'react';

import { FindBookingForm } from '../../components/findBooking';
import { ListBookings } from '../../components/listBooking';
import { DivContainer } from './styles';

interface IBookingsProps {
  findBooking: (findBookingData: IFieldsFormFindBooking) => void;
  listBookingsFinded: Array<IBooking>;
  cancelBooking: (bookId: number) => void;
  isLoading: boolean;
}

export const Booking: React.FC<IBookingsProps> = ({
  findBooking,
  listBookingsFinded,
  cancelBooking,
  isLoading,
}) => {
  return (
    <DivContainer data-testid="book">
      <FindBookingForm onFindBooking={findBooking} />
      <ListBookings
        listBookingsFinded={listBookingsFinded}
        cancelBooking={cancelBooking}
        isLoading={isLoading}
      />
    </DivContainer>
  );
};

Booking.propTypes = {
  findBooking: PropTypes.func.isRequired,
  listBookingsFinded: PropTypes.array.isRequired,
  cancelBooking: PropTypes.func.isRequired,
  isLoading: PropTypes.bool.isRequired,
};
