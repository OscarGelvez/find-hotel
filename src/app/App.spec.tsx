import { IStateMain } from './core/redux/modelo/IStateMain';
import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';

import { IStateBookings } from './core/redux/modelo/IStateBookings';
import { IStateRooms } from './core/redux/modelo/IStateRooms';
import { bookInfo } from './shared/utils/dataBooks';
import { roomInfo } from './shared/utils/data';
import { IBooking } from './feature/Book/models/Booking';
import { IRoom } from './feature/Rooms/models/Room';
import App from './App';
import { GlobalErrorBoundary } from './core/errors/GlobalErrorBoundary';
import { shallow } from 'enzyme';

const mockStore = configureStore([]);

const newBook: IBooking = bookInfo;
const listBookings: IBooking[] = [newBook];

const newRoom: IRoom = roomInfo;
const listRooms: IRoom[] = [newRoom];

const rooms: IStateRooms = {
  listRooms,
  selectedRoomId: -1,
};

const bookings: IStateBookings = {
  listBookings,
  deleteBookingId: -1,
};

const main: IStateMain = {
  isLoading: false,
};

describe('Prueba componente App con redux', () => {
  let store;

  let wrapper;
  beforeEach(() => {
    store = mockStore({
      bookings,
      main,
      rooms,
    });

    wrapper = shallow(
      <GlobalErrorBoundary>
        <Provider store={store}>
          <BrowserRouter>
            <App />
          </BrowserRouter>
        </Provider>
      </GlobalErrorBoundary>
    );
  });
  it('Compara snapshot del componente App renderizado', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
