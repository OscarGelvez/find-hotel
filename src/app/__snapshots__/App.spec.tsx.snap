// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba componente App con redux Compara snapshot del componente App renderizado 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <App />
  </BrowserRouter>
</Provider>
`;
